src/components/Dashboard.tsx

```diff
@@ -33,44 +33,82 @@ const Dashboard: React.FC = () => {
 
   usePageTitle(t('PAGE_TITLE_HOME'))
 
+  const dashboardCategories = templatesByCategory.filter(({ templateCategory: { uiLocation } }) =>
+    uiLocation.includes(UiLocation.Dashboard)
+  )
+
   return (
     <div id="dashboard">
       <Header as="h2" content={t('MENU_ITEM_DASHBOARD')} />
-      {templatesByCategory
-        .filter(({ templateCategory: { uiLocation } }) => uiLocation.includes(UiLocation.Dashboard))
-        .map(({ templates, templateCategory: { icon: categoryIcon, title: categoryTitle } }) => (
-          <div key={categoryTitle} className="template-category">
-            <div className="title">
-              {categoryIcon && <Icon size="large" color="grey" name={categoryIcon} />}
-              <Header as="h4">{categoryTitle}</Header>
-            </div>
-            <div className="templates">
-              {templates.map((template) => (
-                <TemplateComponent key={template.code} template={template} />
-              ))}
-            </div>
-          </div>
+      {dashboardCategories.map(({ templates, templateCategory: { icon, title, code } }) => (
+        <CategoryComponent
+          key={code}
+          templates={templates}
+          icon={icon as SemanticICONS}
+          title={title}
+        />
+      ))}
```

Comment:
Just moved this mapping to its own component, since it needs a little internal logic/state.